packages/ContentAutotagging/Engine/generic/AutotagBaseEngine.ts

```diff
@@ -0,0 +1,745 @@
+                }
+                return AutotagBaseEngine._openai
+            case 'AnthropicLLM':
+                if (!AutotagBaseEngine._anthropic) {
+                    return new AnthropicLLM(process.env['AI_VENDOR_API_KEY__ANTHROPIC'])
+                }
+                return AutotagBaseEngine._anthropic
+            case 'MistralLLM':
+                if (!AutotagBaseEngine._mistral) {
+                    return new MistralLLM(process.env['AI_VENDOR_API_KEY__MISTRALLLM'])
+                }
+                return AutotagBaseEngine._mistral
+            case 'GroqLLM': 
+                if (!AutotagBaseEngine._groq) {
+                    return new GroqLLM(process.env['AI_VENDOR_API_KEY__GROQ'])
+                }
+                return AutotagBaseEngine._groq
+            default:
+                throw new Error(`Model vendor name ${modelDriverClass} not found`)
+        }
+    }
+
+    public async getContentSourceParams(contentSource: ContentSourceEntity, contextUser: UserInfo): Promise<any> {
+        const contentSourceID: string = contentSource.Get('ID')
+        const contentSourceParams = new Map<string, any>()
+
+        const rv = new RunView()
+        const results = await rv.RunView<ContentSourceEntity>({
+            EntityName: 'Content Source Params', 
+            ExtraFilter: `ContentSourceID='${contentSourceID}'`,
+            ResultType: 'entity_object'
+        }, contextUser)
+
+        if (results.Success && results.Results.length) {
+            for (const result of results.Results) {
+                const contentSourceTypeID = result.Get('ContentSourceTypeParamID')
+                const params: ContentSourceTypeParams = await this.getDefaultContentSourceTypeParams(contentSourceTypeID, contextUser)
+                params.contentSourceID = contentSourceID
+
+                if (result.Get('Value')) {
+                    // There is a provided value, so overwrite the default value
+                    params.value = this.castValueAsCorrectType(result.Get('Value'), params.type)
+                }
+                contentSourceParams.set(params.name, params.value)
+            }
+            return contentSourceParams
+        }
+        else {
+            console.log(`No content source params found for content source with ID ${contentSourceID}, using default values`)
+        }
+    }
+
+    public async getDefaultContentSourceTypeParams(contentSourceTypeID: string, contextUser: UserInfo): Promise<ContentSourceTypeParams> {
+        const rv = new RunView()
+        const results = await rv.RunView<ContentSourceEntity>({
+            EntityName: 'Content Source Type Params', 
+            ExtraFilter: `ID='${contentSourceTypeID}'`,
+            ResultType: 'entity_object'
+        }, contextUser)
+
+        if (results.Success && results.Results.length) {
+            const params = new ContentSourceTypeParams()
+            
+            params.name = results.Results[0].Get('Name')
```

Comment:
@nico-oz  **ugly easily breakable code**. set a local variable to `results.Results[0]` which should be strongly typed to the `ContentSourceEntity` class and then use the strongly typed members of that sub-class, you shouldn't be using `Get` and `Set` in nearly all cases.